src/modules/cpu/kernel/tensor_sum.hpp

```diff
@@ -0,0 +1,842 @@
+            srcPtrRow = srcPtrChannel;
+#if __AVX2__
+            __m256d psum = _mm256_setzero_pd();
+#endif
+            for(int i = 0; i < roi.xywhROI.roiHeight; i++)
+            {
+                Rpp32f *srcPtrTemp;
+                srcPtrTemp = srcPtrRow;
+
+                int vectorLoopCount = 0;
+                for (; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrement)
+                {
+#if __AVX2__
+                    __m256d p1[2];
+                    rpp_simd_load(rpp_load8_f32_to_f64_avx, srcPtrTemp, p1);
+                    compute_sum_8_host(p1, &psum);
+#endif
+                    srcPtrTemp += vectorIncrement;
+                }
+                for (; vectorLoopCount < bufferLength; vectorLoopCount++)
+                {
+                    sum += (Rpp64f)(*srcPtrTemp);
+                    srcPtrTemp++;
+                }
+                srcPtrRow += srcDescPtr->strides.hStride;
+            }
+#if __AVX2__
+            rpp_simd_store(rpp_store4_f64_to_f64_avx, sumAvx, &psum);
+#endif
+            for(int i = 0; i < 2; i++)
+                sum += (sumAvx[i] + sumAvx[i + 2]);
+
+            tensorSumArr[batchCount] = (Rpp32f)sum;
+        }
+
+        // Tensor Sum with fused output-layout toggle (NCHW)
+        else if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NCHW))
+        {
+            Rpp64f sum, sumR = 0.0, sumG = 0.0, sumB = 0.0;
+            Rpp64f sumAvxR[4] = {0.0};
+            Rpp64f sumAvxG[4] = {0.0};
+            Rpp64f sumAvxB[4] = {0.0};
+
+            Rpp32f *srcPtrRowR, *srcPtrRowG, *srcPtrRowB;
+            srcPtrRowR = srcPtrChannel;
+            srcPtrRowG = srcPtrRowR + srcDescPtr->strides.cStride;
+            srcPtrRowB = srcPtrRowG + srcDescPtr->strides.cStride;
+#if __AVX2__
+            __m256d psumR = _mm256_setzero_pd();
+            __m256d psumG = _mm256_setzero_pd();
+            __m256d psumB = _mm256_setzero_pd();
+#endif
+            for(int i = 0; i < roi.xywhROI.roiHeight; i++)
+            {
+                Rpp32f *srcPtrTempR, *srcPtrTempG, *srcPtrTempB;
+                srcPtrTempR = srcPtrRowR;
+                srcPtrTempG = srcPtrRowG;
+                srcPtrTempB = srcPtrRowB;
+
+                int vectorLoopCount = 0;
+                for (; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrementPerChannel)
+                {
+#if __AVX2__
+                    __m256d p[6];
+                    rpp_simd_load(rpp_load24_f32pln3_to_f64pln3_avx, srcPtrTempR, srcPtrTempG, srcPtrTempB, p);
+                    compute_sum_24_host(p, &psumR, &psumG, &psumB);
+#endif
+                    srcPtrTempR += vectorIncrementPerChannel;
+                    srcPtrTempG += vectorIncrementPerChannel;
+                    srcPtrTempB += vectorIncrementPerChannel;
+                }
+                for (; vectorLoopCount < bufferLength; vectorLoopCount++)
+                {
+                    sumR += (Rpp64f)(*srcPtrTempR);
```

Comment:
see comment above

Reply:
Done

Reply:
Modified this in all other locations too.

Reply:
Addressed this change.